### PR TITLE
Fix: proto deployement

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**

### CODE REVIEW

**Review by Elli610 — CHANGES_REQUESTED** [review]
Maybe you can importe the const ["MASSA_PROTOFILE_KEY"](https://github.com/massalabs/massa-web3/blob/main/src/web3/SmartContractsClient.ts#L49) from massaweb3 instead of hardcoding 'protoMassa'. In the future, it would be easier to update it if needed.

(I just realized it is not exported in massa-web3. [I proposed a pr to do so](https://github.com/massalabs/massa-web3/pull/383))

**Comment by 0xSwapFeeder** [comment]
> Maybe you can importe the const ["MASSA_PROTOFILE_KEY"](https://github.com/massalabs/massa-web3/blob/main/src/web3/SmartContractsClient.ts#L49) from massaweb3 instead of hardcoding 'protoMassa'. In the future, it would be easier to update it if needed.
> 
> (I just realized it is not exported in massa-web3. [I proposed a pr to do so](https://github.com/massalabs/massa-web3/pull/383))

@gregLibert Is it possible to import anything from massa-web3 into a smart contract ? (in this case a simple constant into the deployer contract ?)

**Comment by gregLibert** [comment]
> @gregLibert Is it possible to import anything from massa-web3 into a smart contract ? (in this case a simple constant into the deployer contract ?)

Nope. the idea for massa web3 is to be typescript compliant. Not assemblyscript. but we do have AS package that can be used dependning on the needs (as, massa-as-sdk...)

**Review by Elli610 — APPROVED** [review]

File: packages/sc-deployer/assembly/contracts/deployer.ts [hunk 1]
```diff
@@ -22,8 +22,7 @@ const CONSTRUCTOR = 'constructor';
 export function main(_: StaticArray<u8>): StaticArray<u8> {
   let masterKey = new StaticArray<u8>(1);
   masterKey[0] = 0x00;
-  let protoKey = new StaticArray<u8>(1);
-  protoKey[0] = 0x02;
+  let protoKey = stringToBytes('protoMassa');
```
Comment (gregLibert) [inline packages/sc-deployer/assembly/contracts/deployer.ts:25]: What is the difference between stringToBytes here and strToBytes below?

File: packages/sc-deployer/src/index.ts [hunk 2]
```diff
@@ -300,15 +300,7 @@ async function deploySC(
...
-      datastore.set(
-        new Uint8Array(
-          new Args()
-            .addU64(BigInt(i + 1))
-            .addUint8Array(u8toByte(2))
-            .serialize(),
-        ),
-        protos, // proto files linked to the contract
-      );
+      datastore.set(strToBytes('protoMassa'), protos);
```
Comment (gregLibert) [inline packages/sc-deployer/src/index.ts:303]: Why aren't you reusing the protoKey constant ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/sc-deployer/assembly/contracts/deployer.ts (hunk 1 above)
Comment (0xSwapFeeder) [reply packages/sc-deployer/assembly/contracts/deployer.ts:25]: There is no usage of strToBytes in the deployer.ts contract (strToBytes is the TS serializer and strintToBytes is the AS serializer)

**Review by gregLibert — APPROVED** [review]

**Comment by gregLibert** [comment]
@0xSwapFeeder can you fix the CI ?

**Comment by 0xSwapFeeder** [comment]
> @0xSwapFeeder can you fix the CI ?

It's done!

**Comment by gregLibert** [comment]
@0xSwapFeeder can you rebase with main please ?

**Pull request merged**